### PR TITLE
UI: Showed host reservations and allocation utilization by task on existing stats charts

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
This applies the new chart patterns to add more information to the existing memory and CPU utilization charts on the client detail, allocation detail, and task detail pages.

## Client Detail

The client detail charts will now show a horizontal annotation (or "high water mark") denoting the [reserved capacity](https://www.nomadproject.io/docs/configuration/client#reserved-resources) for the metric if one has been set.

client detail stats charts with reserved capacity annotations

## Allocation Detail

The allocation detail charts will now break down utilization by task for allocations with multiple tasks.

allocation detail charts with each task utilization stacked on one canvas

## Task Detail

The task detail is unchanged. It shows the same data as the allocation detail charts but narrowed down to the single task.

task detail charts focused on a single task in the same allocation

## Implementation

Since each instance of the previous `PrimaryMetric` component has now deviated, it made sense to refactor it into three components, one for each variation. This resulted in some repetition, but that seemed better than adding even more conditionals and overloading the existing implementation.

There are also some details that arose from using the `StatsTrackers` with Glimmer components.

### CODE REVIEW

File: ui/app/components/primary-metric/node.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { assert } from '@ember/debug';
+import { inject as service } from '@ember/service';
+import { action, get } from '@ember/object';
+
+export default class NodePrimaryMetric extends Component {
+  @service('stats-trackers-registry') statsTrackersRegistry;
+
+  /** Args
+    node = null;
+    metric null; (one of 'cpu' or 'memory'
```
Comment (backspace) [inline ui/app/components/primary-metric/node.js:13]: tiny: missing `)` (repeated elsewhere)

**Review by backspace — CHANGES_REQUESTED** [review]
agreed re the desirability of having an overly-general shared component, breaking it up for the variations makes sense to me.

I found a significant problem in that tooltips are broken for me on the allocation chart, but apart from a few tiny notes, this looks good 😀

File: ui/app/components/primary-metric/allocation.hbs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            <span class="label"><span class="color-swatch swatch-{{this.colorScale}} swatch-{{this.colorScale}}-{{idx}}" />{{series.name}}</span>
+            {{#if (eq this.metric "cpu")}}
+              <span class="value">{{datum.datum.used}} MHz</span>
+            {{else if (eq this.metric "memory")}}
+              <span class="value">{{format-bytes datum.datum.used}}</span>
+            {{else}}
+              <span class="value">{{datum.formatttedY}}</span>
+            {{/if}}
+          </li>
+        </c.Tooltip>
```
[the remark this answers is not in the document]
Comment (DingoEatingFuzz) [reply ui/app/components/primary-metric/allocation.hbs:28]: Hmm. Looks like this happens for allocs with lifecycle tasks that haven't started yet. Is that when you encountered it too?

**Comment by DingoEatingFuzz** [comment]
Okay! I fixed the tooltip bug you caught and another bug with sorting tasks by their lifecycle phase.

**Review by backspace — APPROVED** [review]
nice, indeed fixed for me 🥳

**Pull request merged**